### PR TITLE
Polishing README after v2.0 merge

### PR DESCRIPTION
**Pull request by ahmetalpbalkan**
@kalaninja please take a look and merge if looks good.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,25 +1,32 @@
...
+* Complete lazy evaluation with iterator pattern
+* Supports arrays, slices, maps, strings, channels and custom collections
+(collection needs to implement `Iterable` interface and element - `Comparable`
+interface)
 
 ## Installation
 
     $ go get github.com/ahmetalpbalkan/go-linq
 
+> **warning****warning** `go-linq` has recently introduced _breaking API changes_
```
Comment (kalaninja) [inline README.md]: what's the purpose of writing "warning" two times?

File: README.md [hunk 2]
```diff
@@ -1,25 +1,32 @@
...
+> with v2.0.0. See [release notes](#release-notes) for details. v2.0.0 comes with
+> a refined interface, dramatically increased performance and memory efficiency,
+> and new features such as lazy evaluation ([read more](http://kalan.rocks/2016/07/16/manipulating-data-with-iterators-in-go/)).
+>
+> The old version is still available in `archive/0.9` branch and tagged as `0.9`
+> as well. If you are using `go-linq`, )please vendor a copy of it in your
```
Comment (kalaninja) [inline README.md]: I believe that the bracket before "please" is redundant

**Comment by kalaninja** [comment]
I like the changes that you've done. You definitely write docs much better than I do.

File: README.md (hunk 1 above)
Comment (ahmetalpbalkan) [inline README.md]: oh, I was tryna do ⚠️ ⚠️

File: README.md (hunk 2 above)
Comment (ahmetalpbalkan) [inline README.md]: I'll fix thanks for catching

**Pull request merged**